components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -655,6 +668,74 @@ private static String getComponentCategory(String ref) {
         return "";
     }
 
+    /**
+     * Extract the archive file and validates the openAPI definition
+     *
+     * @param inputStream   file as input stream
+     * @param returnContent whether to return the content of the definition in the response DTO
+     * @return APIDefinitionValidationResponse
+     * @throws APIManagementException if error occurred while parsing definition
+     */
+    public static APIDefinitionValidationResponse extractAndValidateOpenAPIArchive(InputStream inputStream,
+            boolean returnContent) throws APIManagementException {
+        String path = System.getProperty(APIConstants.JAVA_IO_TMPDIR) + File.separator +
+                APIConstants.OPENAPI_ARCHIVES_TEMP_FOLDER + File.separator + UUID.randomUUID().toString();
+        String archivePath = path + File.separator + APIConstants.OPENAPI_ARCHIVE_ZIP_FILE;
+        String extractedLocation = APIFileUtil
+                .extractUploadedArchive(inputStream, APIConstants.OPENAPI_EXTRACTED_DIRECTORY, archivePath, path);
+        File[] listOfFiles = new File(extractedLocation).listFiles();
```

Comment:
you can simply list only the required file.
ex:
```
FilenameFilter filter = (dir, name) -> name.equalsIgnoreCase("swagger.json") || 
name.equalsIgnoreCase("swagger.yaml")
File f = new File(path);
f.list(filter); // should have yaml to json file.
```

Reply:
User can have nested directories and place the API definition inside the leaf directory. Thus when using this filter will only check the root directory and return false .We need to traverse all over the directories and validate for the master swagger

This recursive function is doing that process :
https://github.com/wso2/carbon-apimgt/pull/8509/commits/9ded714b1b9db9c1e33fa3fef5239858920580f9#diff-c212ee576ac1ca0476ff5f5fb61fff6cR672

Reply:
As I mentioned above, we cannot use directory structure and have to stick to naming conversion.